feat!: goodbye `resource` (#2270)
winglang/wing: docs/04-reference/winglang-spec.md

Wing recommends the following formatting and naming conventions:

- Interface names should start with capital letter "I"

> - Class, struct and interface names should be TitleCased
> - Members of classes and interfaces cannot share the same TitleCased

Reviewer: no oxford comma 🥲

```suggestion
- Class, struct, and interface names should be TitleCased
- Members of classes, and interfaces cannot share the same TitleCased
```

there are multiple preflight objects of the same type defined in the same scope, they
must all have an explicit id.

Preflight objects instantiated at block scope root level of entrypoint are assigned the
root app as their default implicit scope.

Preflight object instantiation syntax uses the `let` keyword the same way variables are declared in
Wing. The `as` and `in` keywords can be used to customize the scope and identifier assigned to this
preflight object.

```pre
let <name>[: <type>] = new <Type>(<args>) [as <id>] [in <scope>];
```

```TS
// Wing Code:
let a = Foo(); // with default scope and id
let a = Foo() in scope; // with user-defined scope
let a = Foo() as "custom-id" in scope; // with user-defined scope and id
let a = Foo(...) as "custom-id2" in scope; // with constructor arguments
```

"id" must be of type string. It can also be a string literal with substitution
support (normal strings as well as shell strings).  
"scope" must be an expression that resolves to a preflight object.

Preflight objects can be captured into inflight scopes and once that happens, inside
the capture block only the inflight members are available.

> Preflight classes can extend other preflight classes (but not structs) and implement preflight interfaces.

Author: Added to the interfaces section... We need to rewrite this doc, but for now it's okay I think.

### 4.2 Exports

In preflight, anything with `public` at block scope level is importable.
This includes functions, classes, structs and interfaces.
In inflight, the above excluding preflight classes are importable.
Variables are not exportable.

> Preflight classes are not usable in inflight functions. There is no synthesizer inside
> and no deployment system in the inflight body to provision.

Reviewer: Yeah "usable" seems too broad here. I assume this was the intended meaning?

```suggestion
Preflight classes cannot be instantiated in inflight functions. There is no synthesizer inside
and no deployment system in the inflight body to provision.
```
